qutebrowser/mainwindow/statusbar/command.py

```diff
@@ -232,6 +232,11 @@ def keyPressEvent(self, e):
         Enter/Shift+Enter/etc. will cause QLineEdit to think it's finished
         without command_accept to be called.
         """
+        text = self.text()
+        if text in modeparsers.STARTCHARS and e.key() == Qt.Key_Backspace:
+            modeman.leave(self._win_id, usertypes.KeyMode.command,
+                          'prefix deleted')
```

Comment:
You should also call `e.accept()` and `return` so the key press isn't processed further (as we're leaving insert mode anyways).